route.go

```diff
@@ -130,11 +132,9 @@ func (r *Route) parse(rw http.ResponseWriter, req *http.Request) bool {
 				}
 			}
 		}
-		if r.Match(req) {
-			r.Handler.ServeHTTP(rw, req)
-			vars.Lock()
-			delete(vars.v, req)
-			vars.Unlock()
+
+		if ok, vars := r.Match(req); ok {
+			r.serveMatchedRequest(rw, req, vars)
```

Comment:
This functionality used to be inline, but I needed to extract it to a method to allow different code to run on <go1.7.